source/VMelnalksnis.PaperlessDotNet/Documents/IDocumentClient.cs

```diff
@@ -48,6 +49,13 @@ public interface IDocumentClient
 	/// <returns>The document with the specified id if it exists; otherwise <see langword="null"/>.</returns>
 	Task<Document<TFields>?> Get<TFields>(int id, CancellationToken cancellationToken = default);
 
+	/// <summary>Downloads the file of the document.</summary>
+	/// <param name="id">The id of the document to download.</param>
+	/// <param name="original">Download the original document when an archived document is available.</param>
+	/// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+	/// <returns>The content of the document.</returns>
+	Task<DocumentContent> Download(int id, bool original, CancellationToken cancellationToken = default);
```

Comment:
Can you maybe also add methods for getting the preview and the thumbnail as well? I think the  only difference should be the request uri.

Reply:
I have added the methods as well but I am not sure if it is useful to have the download methods for the preview. The only difference as far as i can see was the `DispositionType` of the `ContentDispositionHeader`.

In the case of downloading the original it is set to `attachment` and when downloading the preview it is set to `inline`. As far as i know this header indicates if the document should be diplayed or downloaded directly:   https://developer.mozilla.org/de/docs/Web/HTTP/Headers/Content-Disposition

Do you think it is useful to provide both methods?

Reply:
Ah I guess not, should have read the documentation with more attention.

Reply:
I think i will just leave the methods as they are, even if they return almost the same thing.
If the calls change in the future we have already implemented those.